zerver/context_processors.py

```diff
@@ -44,7 +44,10 @@ def get_realm_from_request(request: HttpRequest) -> Realm:
         # need to do duplicate queries on the same realm while
         # processing a single request.
         subdomain = get_subdomain(request)
-        request.realm = get_realm(subdomain)
+        try:
+            request.realm = get_realm(subdomain)
+        except Realm.DoesNotExist:
+            request.realm = None
```

Comment:
This function either has the wrong type annotation or should reraise an exception...

Reply:
Fixed the type.